### ISSUE TITLE
Infill don't respect the perimeter boundaries 

### ISSUE THREAD

**Issue by thiagopeixoto16** [alexrj/Slic3r#4381]
### Version
Slic3r-master.2018.04.13.1651.df993c5.64bit

### Operating system type + version
Windows 10

### Behavior
In some cases the infill (specifically the inner support of top layers) overlap the perimeter generating that:
![image](https://user-images.githubusercontent.com/4227271/38769705-a8934270-3fdd-11e8-93ba-e008ae5f7c3f.png)

I found no similar issue, if there is one please apologize me, close this and I'll contribute on the main tread.

#### STL/Config (.ZIP) where problem occurs
[Issue.zip](https://github.com/slic3r/Slic3r/files/1910544/Issue.zip)

**Comment by lordofhyphens** [comment]
Looks like it has something to do with the number of top solid layers and that specific slope right there. 

With your config it happens at Z=138.60 according to the preview.

**Comment by thiagopeixoto16** [comment]
Yes and there are some others on other layers from this same STL

**Comment by lordofhyphens** [comment]
Dropping top layers to 3 will let you avoid the bug for now.

I think that can be done with modifier meshes.
